crates/backend/src/local.rs

```diff
@@ -328,26 +348,43 @@ impl WriteBackend for LocalBackend {
     ///
     /// # Errors
     ///
-    /// * [`LocalBackendErrorKind::OpeningFileFailed`] - If the file could not be opened.
+    /// * [`LocalBackendErrorKind::OpeningFileForWritingFailed`] - If the file could not be opened for writing.
+    /// * [`LocalBackendErrorKind::DirectoryCreationFailed`] - If the parent directory could not be created.
     /// * [`LocalBackendErrorKind::FromTryIntError`] - If the length of the bytes could not be converted to u64.
     /// * [`LocalBackendErrorKind::SettingFileLengthFailed`] - If the length of the file could not be set.
     /// * [`LocalBackendErrorKind::CouldNotWriteToBuffer`] - If the bytes could not be written to the file.
     /// * [`LocalBackendErrorKind::SyncingOfOsMetadataFailed`] - If the metadata of the file could not be synced.
     ///
-    /// [`LocalBackendErrorKind::OpeningFileFailed`]: LocalBackendErrorKind::OpeningFileFailed
+    /// [`LocalBackendErrorKind::OpeningFileForWritingFailed`]: LocalBackendErrorKind::OpeningFileForWritingFailed
     /// [`LocalBackendErrorKind::FromTryIntError`]: LocalBackendErrorKind::FromTryIntError
     /// [`LocalBackendErrorKind::SettingFileLengthFailed`]: LocalBackendErrorKind::SettingFileLengthFailed
     /// [`LocalBackendErrorKind::CouldNotWriteToBuffer`]: LocalBackendErrorKind::CouldNotWriteToBuffer
     /// [`LocalBackendErrorKind::SyncingOfOsMetadataFailed`]: LocalBackendErrorKind::SyncingOfOsMetadataFailed
+    /// [`LocalBackendErrorKind::DirectoryCreationFailed`]: LocalBackendErrorKind::DirectoryCreationFailed
     fn write_bytes(&self, tpe: FileType, id: &Id, _cacheable: bool, buf: Bytes) -> Result<()> {
         trace!("writing tpe: {:?}, id: {}", &tpe, &id);
         let filename = self.path(tpe, id);
+
+        // create parent directory if it does not exist
+        if let Some(parent) = filename.parent() {
```

Comment:
I would make a method `parent_path` and then just use `self.parent_path(tpe,id)` here.

Reply:
Added!

Reply:
Actually I thought about having something along the `fn path`, but just without the last `join`. We know there is a parent because we just create paths which do have a parent. My idea was just about removing the error handling which will never be called.